### PR TITLE
Make pro connector letters undated and name PDF files by recipient

### PR DESCRIPTION
**Pull request by holdenk**
## Summary
Updates the pro connector letter template and generation to remove the date from printed letters and use the recipient's name and organization as the document title (which browsers use as the default PDF file name when saving).

## Changes

- **Remove date from letters**: Letters are printed in batches and mailed later, so including a date only causes it to go stale. Removed the `<div class="date">` from the template and the `today` context variable from the view.

- **Add `build_letter_document_title()` function**: New helper that generates a descriptive document title like "Letter - Dr. Jane Smith - Acme Health Clinic" for use as the browser's print-to-PDF file name. Includes:
  - Sanitization of illegal/awkward file name characters (`\/:*?"<>|`)
  - Whitespace normalization and collapsing
  - Per-part length capping (60 chars) to prevent unwieldy file names
  - Deduplication when business name matches person's name
  - Fallback to bare "Letter" when neither name nor organization is known
  - Prevention of trailing dots (invalid in Windows file names)

- **Update letter view**: Changed the page title from the static "Pro Connector Letter" to the dynamic result of `build_letter_document_title(pro)`, so staff printing batches get distinguishable files instead of a stack of identically named PDFs.

- **Enhance letterhead styling**: Added flexbox layout to the letterhead to accommodate a logo image alongside the organization name and contact info, with print-color-adjust to preserve colors when printing.

- **Update template instructions**: Clarified in the print instructions that saving as PDF will use the document title as the file name.

## Testing
Added comprehensive test coverage in `LetterDocumentTitleTest` with 9 test cases covering:
- Basic title generation with name and organization
- Absence of dates in the title
- Handling of missing name or organization
- Deduplication of matching names
- Illegal character removal
- Whitespace normalization and truncation
- Edge cases like trailing dots and organizations named "Letter"

Also added integration tests in `LetterViewTest` to verify the title appears in the rendered HTML and that no date is present in the response.

https://claude.ai/code/session_014W8kJtmggZwPLB7CLDXac1

## Summary by CodeRabbit

* **New Features**
  * Printable letters now include a professional letterhead logo.
  * PDF filenames are automatically generated from recipient and organization names.
  * Names are cleaned, deduplicated, and formatted for safe, readable filenames, with a fallback title when details are unavailable.
  * Print instructions explain that saving as a PDF uses the document title.

* **Updates**
  * Removed the printed date from printable letters.

### CODE REVIEW

File: fighthealthinsurance/templates/proconnector_letter.html [hunk 1]
```diff
@@ -202,15 +211,16 @@
 
   <main class="sheet">
     <div class="letterhead">
-      <div class="org">Fight Health Insurance</div>
-      <div class="contact">
-        <a href="https://www.fighthealthinsurance.com">www.fighthealthinsurance.com</a>
-        &middot; {{ contact_email }}
+      <img src="{% static 'images/better-logo-optimized.png' %}" alt="">
```
[the remark this answers is not in the document]
Comment (holdenk) [reply fighthealthinsurance/templates/proconnector_letter.html:214]: Verified this is a false positive. Since Django 3.1, a relative `STATIC_URL` is passed through `SETTINGS_TO_ADD_SCRIPT_PREFIX` handling, which prepends the script prefix — so `STATIC_URL = "static/"` resolves to `/static/` at runtime and `{% static %}` renders the root-relative `/static/images/better-logo-optimized.png`. Confirmed both in a shell (`settings.STATIC_URL` → `'/static/'`) and by rendering the letter page at `/timbit/help/proconnector/1/letter` in a browser, where the logo loads in the page and in the printed PDF.

---
_Generated by [Claude Code](https://claude.ai/code)_
